core/primitives/piece/piece_data.hpp

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_PIECE_DATA_HPP
+#define CPP_FILECOIN_PIECE_DATA_HPP
+
+#include <string>
+
+namespace fc::primitives::piece {
+
+  class PieceData {
+   public:
+    explicit PieceData(const std::string &path_to_file);
```

Comment:
Add move constructor to prevent double close.

Reply:
Added